### DISCUSSION TITLE
Any sample how to replace one ChromiumWebBrowser object with another into the same form

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3580] [Q&A]
I want to create Winforms ChromiumWebBrowser object (let`s call it objectA), put it into the form and then replace it with another ChromiumWebBrowser object (objectB). I wanna do that without disposing objectA (so that I would be able later again replace objectB with objectA). Is that possible? Any existing sample that could help me?

**Comment by amaitland** [comment]
> I wanna do that without disposing objectA (so that I would be able later again replace objectB with objectA). Is that possible?

Are you using `WinForms` or `WPF?

Just remove the `ChromiumWebBrowser` instance from it's parent and add the new `ChromiumWebBrowser` instance. Once you've removed the old instance make sure you call `Dispose` to free all the resources.

**Reply by assopri** [reply]
Thank you! I`m using winforms.

**Reply by amaitland** [reply]
Generally it should look something like the following:

```c#
//Parent in this example is a toolStripContainer's ContentPanel
//Can be any parent really.
var parent = toolStripContainer.ContentPanel;
var oldBrowser = browser;
//Remove old browser
parent.Controls.Remove(oldBrowser);
//Create new instance
browser = new ChromiumWebBrowser();
//Add to parent
parent.Controls.Add(browser);

//Dispose of old Brower
oldBrowser.Dispose();
```

**Reply by assopri** [reply]
Thank you! I'm using Winforms. Can I avoid disposing oldbrowser? Because I want to use it later (for instance I could have authorized using it on the website and I don`t want to repeat that process). Actually this is the question - can I reuse browser instance after detaching it from parent and subsequent comeback to the same parent?

**Reply by amaitland** [reply]
> Actually this is the question - can I reuse browser instance after detaching it from parent and subsequent comeback to the same parent?

Long as the `Control.Handle` remains valid that should be fine.

If for some reason the `Handle` is recreated then you'll need to set [ParkControlOnHandleDestroyed = true](http://cefsharp.github.io/api/90.6.x/html/P_CefSharp_WinForms_ChromiumWebBrowser_ParkControlOnHandleDestroyed.htm)
